src/Examples/Tools/DegeneracyHunter/degeneracy_hunter.ipynb

```diff
@@ -0,0 +1,1538 @@
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      "  10  4.1177584e+01 1.30e-03 8.56e-03  -2.5 1.48e-01    -  1.00e+00 1.00e+00f  1\n",
+      "  11  4.1157407e+01 7.91e-06 1.09e-05  -3.8 9.88e-03    -  1.00e+00 1.00e+00h  1\n",
+      "  12  4.1157067e+01 2.59e-10 1.57e-09  -5.7 1.30e-04    -  1.00e+00 1.00e+00h  1\n",
+      "  13  4.1157063e+01 2.24e-14 1.63e-13  -8.6 1.52e-06    -  1.00e+00 1.00e+00h  1\n",
+      "\n",
+      "Number of Iterations....: 13\n",
+      "\n",
+      "                                   (scaled)                 (unscaled)\n",
+      "Objective...............:   4.1157063321491172e+01    4.1157063321491172e+01\n",
+      "Dual infeasibility......:   1.6278818285040336e-13    1.6278818285040336e-13\n",
+      "Constraint violation....:   2.2426505097428162e-14    2.2426505097428162e-14\n",
+      "Complementarity.........:   2.5065697306197698e-09    2.5065697306197698e-09\n",
+      "Overall NLP error.......:   2.5065697306197698e-09    2.5065697306197698e-09\n",
+      "\n",
+      "\n",
+      "Number of objective function evaluations             = 24\n",
+      "Number of objective gradient evaluations             = 14\n",
+      "Number of equality constraint evaluations            = 24\n",
+      "Number of inequality constraint evaluations          = 24\n",
+      "Number of equality constraint Jacobian evaluations   = 14\n",
+      "Number of inequality constraint Jacobian evaluations = 14\n",
+      "Number of Lagrangian Hessian evaluations             = 13\n",
+      "Total CPU secs in IPOPT (w/o function evaluations)   =      0.003\n",
+      "Total CPU secs in NLP function evaluations           =      0.000\n",
+      "\n",
+      "EXIT: Optimal Solution Found.\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "{'Problem': [{'Lower bound': -inf, 'Upper bound': inf, 'Number of objectives': 1, 'Number of constraints': 3, 'Number of variables': 5, 'Sense': 'unknown'}], 'Solver': [{'Status': 'ok', 'Message': 'Ipopt 3.13.2\\\\x3a Optimal Solution Found', 'Termination condition': 'optimal', 'Id': 0, 'Error rc': 0, 'Time': 0.016573190689086914}], 'Solution': [OrderedDict([('number of solutions', 0), ('number of solutions displayed', 0)])]}"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "opt.options['max_iter'] = 50\n",
+    "opt.solve(m, tee=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As expected, Ipopt has no trouble solving this optimization problem."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Check if any constraint residuals are large than 1E-14"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's now inspect the new solution to see which (if any) constraints have residuals larger than 10$^{-14}$."
```

Comment:
This might just be Github's online renderer, but the $^{-14}$ did not appear correctly when reviewing the PR. Just check to make sure it renders correctly in an actual Jupyter interface.

Reply:
It renders correctly.